src/DependencyInjection/Configuration.php

```diff
@@ -39,13 +39,49 @@ public function getConfigTreeBuilder(): TreeBuilder
                         ->arrayPrototype()
                             ->children()
                                 ->scalarNode('url')->info('URL of the hub\'s publish endpoint')->example('https://demo.mercure.rocks/.well-known/mercure')->end()
-                                ->scalarNode('jwt')->info('JSON Web Token to use to publish to this hub.')->end()
-                                ->scalarNode('jwt_provider')->info('The ID of a service to call to generate the JSON Web Token.')->end()
+                                ->scalarNode('public_url')->info('URL of the hub\'s public endpoint')->example('https://demo.mercure.rocks/.well-known/mercure')->defaultNull()->end()
+                                ->arrayNode('jwt')
+                                    ->beforeNormalization()
+                                        ->ifString()
+                                        ->then(static function (string $token): array {
+                                            return [
+                                                'value' => $token,
+                                            ];
+                                        })
+                                    ->end()
+                                    ->info('JSON Web Token configuration.')
+                                        ->children()
+                                            ->scalarNode('value')->info('JSON Web Token to use to publish to this hub.')->end()
+                                            ->scalarNode('provider')->info('The ID of a service to call to provide the JSON Web Token.')->end()
+                                            ->scalarNode('factory')->info('The ID of a service to call to create the JSON Web Token.')->end()
+                                            ->arrayNode('publish')
+                                                ->beforeNormalization()->castToArray()->end()
+                                                ->scalarPrototype()->end()
+                                                ->info('A list of topics to allow publishing to when using the given factory to generate the JWT.')
+                                            ->end()
+                                            ->arrayNode('subscribe')
```

Comment:
Maybe should we allow to pass some default extra claims here (such as `iss`)?

Reply:
additional claims are configured via the `create()` method, until we want to allow both configuring additiona claims both in the constructor and the `create()` method.